src/components/Article.jsx

```diff
@@ -18,6 +19,25 @@ const Article = () => {
         });
     }, []);
 
+    const updateVoteDisplay = (value) => {
+        setArticle((originalArticle) => {
+            const newArticle = {...originalArticle};
+            newArticle.votes = newArticle.votes + value;
+            return newArticle;
+        });
+    }
+
+    const upVote = () => {
+        if (allowVote){
+            updateVoteDisplay(1);
+            setAllowVote(false);
+            upVoteArticle(article_id).catch(() => {
+                updateVoteDisplay(-1);
```

Comment:
Ideally if the patch fails, we want to give the user some sort of feedback to state it has infact failed. This may be a message saying 'not able to upvote at this time'